packages/expo-firebase-core/src/FirebaseCore.ts

```diff
@@ -43,11 +48,12 @@ export const DEFAULT_APP_NAME: string | void = ExpoFirebaseCore
  * }
  * ```
  */
-export const DEFAULT_APP_OPTIONS: IFirebaseOptions | void = ExpoFirebaseCore
+export const DEFAULT_APP_OPTIONS: Partial<IFirebaseOptions> | void = ExpoFirebaseCore
   ? ExpoFirebaseCore.DEFAULT_APP_OPTIONS
   : undefined;
 
+// @needsAudit
 /**
  * The default Firebase options as defined in `web.config.firebase` in `app.json`.
  */
-export const DEFAULT_WEB_APP_OPTIONS: IFirebaseOptions | void = getDefaultWebOptions();
+export const DEFAULT_WEB_APP_OPTIONS: Partial<IFirebaseOptions> | void = getDefaultWebOptions();
```

Comment:
This is a change to the types, we should land this in a separate PR and add a changelog entry for it

Reply:
Usually the types changes required for the automatic API generation have landed with the package conversion.

I will update the changelog, however I think that this is not that important change for the users, because as I mentioned in the description I have only moved the `Partial` util placing. Would you like to see additional type instead? 

Also TBH, it's the first usage of `Partial` in Expo types I have seen in last 5 months, and it could be easily removed in favor of optional markings for `IFirebaseOptions`.

Reply:
> Usually the types changes required for the automatic API generation have landed with the package conversion.

Okay sounds good, just checking 👍 

> I will update the changelog, however I think that this is not that important change for the users, because as I mentioned in the description I have only moved the Partial util placing. Would you like to see additional type instead, just for the clarity?
>
> Also TBH, it's the first usage of Partial in Expo types I have seen in last 5 months, and it could be easily removed in favor of optional markings for IFirebaseOptions.

Ah I see, I had missed that when reading the PR description, now I see.

As we're changing it from an `interface` to a `type` (which I think is a good thing and it should probably have been a type from the beginning), still keeping the `I` prefix feels wrong here. It will be confusing and I think we need a better migration strategy.

I would suggest to rename `IFirebaseOptions` to `FirebaseOptions` and introduce a new `IFirebaseOptions` type which is marked as deprecated, and which simply refers to `FirebaseOptions`. This way older projects will keep working and we move forward cleaner (imo).

And this should be mentioned in the changelog, as well as `expo-firebase-analytics` and `expo-firebase-recaptcha` should be updated to use the new type (I can help with that).

```
export type FirebaseOptions = {
   ///
};

/**
@deprecated ...
**/
export type IFirebaseOptions = FirebaseOptions;
```

Reply:
Hmm, there were in the past bigger and breaking changes to the types, but we did not introduce any deprecations so far (just replaced types with not in the changelog):
* https://github.com/expo/expo/pulls?q=is%3Apr+author%3ASimek+doc+comments+is%3Aclosed

It feels like adding deprecation here a bit too much, and an introduction of the additional complexity which we do not need.

There are few things which I want refactor, but so far I was just leaving the tasks on Linear for that, example:
* https://linear.app/expo/issue/ENG-1738/[brightness]-rename-usesystembrightnessasync

Also from the experience from converting other modules I can tell that patterns seen in this package (like `I` in name for the interface), is not propagated across the workspace. It might be something worth discussing, the more consistent approach to naming sounds good, but already converting modules requires a large amount of work, and I just tiring to minimize the changeset size as possible.

Reply:
> Hmm, there were in the past bigger and breaking changes to the types, but we did not introduce any deprecations so far (just replaced types, usually with note in the changelog):
> It feels like adding deprecation here a bit too much, and an introduction of the additional complexity which we do not need.

Okay that sounds reasonable. In that case I would like to opt for removing the `I` and fix it once and for all.